internal/pkg/template/templates/workloads/services/static-site/cf.yml

```diff
@@ -42,6 +42,67 @@ Resources:
         Rules:
           - ObjectOwnership: BucketOwnerEnforced
 
+  BucketPolicyForCloudFront:
+    Metadata:
+      'aws:copilot:description': 'A bucket policy to grant CloudFront read access to the Static Site bucket'
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref Bucket
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Sid: AllowCloudFrontServicePrincipalReadOnly
+            Effect: Allow
+            Principal:
+              Service: cloudfront.amazonaws.com
+            Action: s3:GetObject
+            Resource:
+              - !Sub
+                - arn:${AWS::Partition}:s3:::${bucket}
+                - bucket: !Ref Bucket
+              - !Sub
+                - arn:${AWS::Partition}:s3:::${bucket}/*
+                - bucket: !Ref Bucket
+            Condition: 
+              StringEquals:
+                AWS:SourceArn: !Sub
+                  - arn:${AWS::Partition}:cloudfront::${AWS::AccountId}:distribution/${cfDistributionID}
+                  - cfDistributionID: !Ref CloudFrontDistribution
+
+  CloudFrontOriginAccessControl:
+    Metadata:
+      'aws:copilot:description': 'Access control to make the content in the S3 bucket only accessible through CloudFront'
+    Type: AWS::CloudFront::OriginAccessControl
+    Properties: 
+      OriginAccessControlConfig: 
+          Description: Access control for static s3 origin
+          Name: !Sub 'copilot-${AppName}-${EnvName}-${WorkloadName}-origin-access-control'
```

Comment:
I couldn't find if there is any limitation on the length of the `Name` field from the docs but would you mind testing out a very long name to figure out the limit? and maybe we can truncate accordingly if needed